orchestrator/careplanservice/handle_get_test.go

```diff
@@ -0,0 +1,727 @@
+			id:           "1",
+			resourceType: "Condition",
+			returnedResource: &fhir.Condition{
+				Id: to.Ptr("1"),
+				Subject: fhir.Reference{
+					Identifier: &fhir.Identifier{
+						System: to.Ptr("http://fhir.nl/fhir/NamingSystem/bsn"),
+						Value:  to.Ptr("123456789"),
+					},
+				},
+			},
+			errorFromPatientBundleRead: errors.New("error"),
+			expectError:                true,
+		},
+		{
+			ctx:          auth.WithPrincipal(context.Background(), *auth.TestPrincipal3),
+			name:         "Condition exists, subject is patient, patient returned, incorrect principal",
+			id:           "1",
+			resourceType: "Condition",
+			returnedResource: &fhir.Condition{
+				Id: to.Ptr("1"),
+				Subject: fhir.Reference{
+					Identifier: &fhir.Identifier{
+						System: to.Ptr("http://fhir.nl/fhir/NamingSystem/bsn"),
+						Value:  to.Ptr("123456789"),
+					},
+				},
+			},
+			returnedCarePlanBundle: &fhir.Bundle{
+				Entry: []fhir.BundleEntry{
+					{
+						Resource: carePlan1Raw,
+					},
+					{
+						Resource: careTeam2Raw,
+					},
+				},
+			},
+			returnedPatientBundle: &fhir.Bundle{
+				Entry: []fhir.BundleEntry{
+					{
+						Resource: patient1Raw,
+					},
+				},
+			},
+			expectError: true,
+		},
+		{
+			ctx:          auth.WithPrincipal(context.Background(), *auth.TestPrincipal1),
+			name:         "Condition exists, subject is patient, patient returned, correct principal",
+			id:           "1",
+			resourceType: "Condition",
+			returnedResource: &fhir.Condition{
+				Id: to.Ptr("1"),
+				Subject: fhir.Reference{
+					Identifier: &fhir.Identifier{
+						System: to.Ptr("http://fhir.nl/fhir/NamingSystem/bsn"),
+						Value:  to.Ptr("123456789"),
+					},
+				},
+			},
+			returnedCarePlanBundle: &fhir.Bundle{
+				Entry: []fhir.BundleEntry{
+					{
+						Resource: carePlan1Raw,
+					},
+					{
+						Resource: careTeam2Raw,
+					},
+				},
+			},
+			returnedPatientBundle: &fhir.Bundle{
+				Entry: []fhir.BundleEntry{
+					{
+						Resource: patient1Raw,
+					},
+				},
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range taskTests {
+		testHelperHandleGetResource[fhir.Task](t, tt)
+	}
+	for _, tt := range carePlanTests {
+		testHelperHandleGetResource[fhir.CarePlan](t, tt)
+	}
+	for _, tt := range careTeamTests {
+		testHelperHandleGetResource[fhir.CareTeam](t, tt)
+	}
+	for _, tt := range patientTests {
+		testHelperHandleGetResource[fhir.Patient](t, tt)
+	}
+	for _, tt := range questionnaireTests {
+		testHelperHandleGetResource[fhir.Questionnaire](t, tt)
+	}
+	for _, tt := range questionnaireResponseTests {
+		testHelperHandleGetResource[fhir.QuestionnaireResponse](t, tt)
+	}
+	for _, tt := range serviceRequestTests {
+		testHelperHandleGetResource[fhir.ServiceRequest](t, tt)
+	}
+	for _, tt := range conditionTests {
+		testHelperHandleGetResource[fhir.Condition](t, tt)
+	}
+}
+
+func testHelperHandleGetResource[T any](t *testing.T, params TestStruct[T]) {
```

Comment:
I think this could be simplified by turning it inside-out: just make actual Test_XX (e.,g. Task, Patient, etc) functions (or tests within the test with `t.Run()`), and having this setup functionality called from each respective test. Since the logic for testing a particular resource is now spread over 2 spots (test case definition and handler calling in this function).

Reply:
I have refactored the tests into their own methods and simplified the unit test cases a bit, it should be cleaner and easier to look over now